lab-geoff/lib/storage.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const uuid = require('node-uuid');
+const Promise = require('bluebird');
+const createError = require('http-errors');
+const debug = require('debug')('mnp:storage');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+module.exports = exports = {};
+
+exports.createItem = function(collection, item) {
+  debug('createItem()', item);
+  if(!collection) return Promise.reject(createError(400, 'collection name not supplied'));
+  if(!item) return Promise.reject(createError(400, 'missing item to create'));
+
+  item.id = item.id || uuid.v4().slice(0,8);
```

Comment:
Checkout the `shortid` npm module. I often use it instead of uuid's because they are prettier. They do have a higher chance of collision though.